File: spring-boot3/admin/src/main/java/com/admin/genTable/entity/GenTableColumn.java
package com.admin.genTable.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;
import lombok.experimental.Accessors;

import java.time.LocalDateTime;

@Data
@TableName("gen_table_column")
@Accessors(chain = true)
public class GenTableColumn {
    /** 主键ID */
    @TableId(type = IdType.ASSIGN_ID)
    private String id;

    /** 所属表ID */
    private String tableId;

    /** 列名 */
    private String columnName;

    /** 列描述 */
    private String columnComment;

    /** 列类型（如 varchar(255)、int 等） */
    private String columnType;

    /** Java类型（如 java.lang.String、Integer） */
    private String javaType;

    /** Java字段名（如 userName） */
    private String javaField;

    /** 是否主键，1是，0否 */
    private Boolean isPk;

    /** 是否自增，1是，0否 */
    private Boolean isIncrement;

    /** 是否必填，1是，0否 */
    private Boolean isRequired;

    /** 是否插入字段，1是，0否 */
    private Boolean isInsert;

    /** 是否编辑字段，1是，0否 */
    private Boolean isEdit;

    /** 是否列表字段，1是，0否 */
    private Boolean isList;

    /** 是否查询字段，1是，0否 */
    private Boolean isQuery;

    /** 查询方式（如 EQ、LIKE、BETWEEN） */
    private String queryType;

    /** 表单控件类型（如 input、select、datetime） */
    private String htmlType;

    /** 字段排序 */
    private Integer sort;

    /** 创建时间 */
    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createTime;

    /** 更新时间 */
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updateTime;
}

File: ant-design-pro/src/services/ant-design-pro/genTableColumn.ts
import { request } from '@umijs/max';
import genTableForm from '@/pages/tool/gen/compoents/GenTableForm';

export type R<T = any> = {
  code: number;
  success: boolean;
  result: T;
  message?: string;
};

export type GenTableColumn = {
  id?: string;
  tableId?: string;
  columnName?: string;
  columnComment?: string;
  columnType?: string;
  javaType?: string;
  javaField?: string;
  isPk?: boolean;
  isIncrement?: boolean;
  isRequired?: boolean;
  isInsert?: boolean;
  isEdit?: boolean;
  isList?: boolean;
  isQuery?: boolean;
  queryType?: string;
  htmlType?: string;
  sort?: number;
  createTime?: string;
  updateTime?: string;
};

/** 查询字段列表 GET /genTableColumn/list */
export async function listGenTableColumn(
  params: {
    genTableColumn: GenTableColumn;
    pageNo: number;
    pageSize: number;
  },
  options?: { [key: string]: any },
) {
  const { genTableColumn, pageNo, pageSize } = params;
  return request<R>('/api/genTableColumn/list', {
    method: 'GET',
    params: {
      ...genTableColumn,
      pageNo,
      pageSize,
    },
    ...(options || {}),
  });
}
